### ISSUE TITLE
Importing .jar file into Flutter application and using it with platform channel

### ISSUE THREAD

**Issue by Iftach-Nevo** [flutter/flutter#81273]
I am trying to import a .jar file into flutter so I can call functions from there via method channel. I find this to be a quite difficult taks. I tried going through the two guides as shown here #16259 but im getting errors on build.

When adding implementation fileTree(include: '*.jar', dir: 'libs') to the build.gradle I get this error:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/Iftach/Desktop/app/call_c_java_cpp/android/build.gradle' line: 11

* What went wrong:
A problem occurred evaluating root project 'android'.
> Could not find method implementation() for arguments [directory 'libs'] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 706ms
Exception: Gradle task assembleDebug failed with exit code 1
```

I tried opening the android module in android studio and import the .jar file from there and I still get errors on builds.

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeDebugJavaResource'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Failed to transform flutterJava.jar to match attributes {artifactType=android-java-res, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for JetifyTransform: /Users/Iftach/Desktop/app/flutter_app_test_java/android/app/libs/flutterJava.jar.
         > Failed to transform '/Users/Iftach/Desktop/app/flutter_app_test_java/android/app/libs/flutterJava.jar' using Jetifier. Reason: IllegalArgumentException, message: Unsupported class file major version 59. (Run with --stacktrace for more details.)
           Suggestions:
            - Check out existing issues at https://issuetracker.google.com/issues?q=componentid:460323&s=modified_time:desc, it's possible that this issue has already been filed there.
            - If this issue has not been filed, please report it at https://issuetracker.google.com/issues/new?component=460323 (run with --stacktrace and provide a stack trace if possible).

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
Exception: Gradle task assembleDebug failed with exit code 1
```

Integrating my own .jar file into flutter is essential for my app, so I really hope there is a way I can do this.

Thank you.

**Comment by TahaTesser** [comment]
Hi @Iftach-Nevo 
Here I created a [sample repositor](https://github.com/tesserx/flutter-android-jar-sample)y that uses a native android library `Glide` (for loading network images in native android) 
`git clone https://github.com/tesserx/flutter-android-jar-sample.git`
`cd flutter-android-jar-sample`
`flutter run -v`

When the app launches, tap `Launch native activity`, this will lead to native android activity and loads an image which `Glide`

<details> 
<summary>flutter doctor -v</summary> 

```console
[✓] Flutter (Channel stable, 2.0.5, on Microsoft Windows [Version 10.0.19042.928], locale en-US)
    • Flutter version 2.0.5 at C:\Users\Taha\Code\flutter_stable
    • Framework revision adc687823a (11 days ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = C:\Users\Taha\Code\SDK
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.9.4)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.9.31205.134
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] IntelliJ IDEA Community Edition (version 2021.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.3.3
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.55.2)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • Windows (desktop)                  • windows       • windows-x64    • Microsoft Windows [Version 10.0.19042.928]
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 90.0.4430.93
    • Edge (web)                         • edge          • web-javascript • Microsoft Edge 90.0.818.46

! Doctor found issues in 1 category.
```
		
</details>

**Comment by Iftach-Nevo** [comment]
Hi @TahaTesser, Thank you for your reply
I ran the sample repository and it works find, when I add my .jar it throws an error tho:

```
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Execution failed for task ':app:compileDebugKotlin'.                    
> Could not resolve all files for configuration ':app:debugCompileClasspath'.
   > Failed to transform flutter-java.jar to match attributes {artifactType=android-classes-jar, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for JetifyTransform: /Users/Iftach/Desktop/github_sample/flutter-android-jar-sample/android/app/libs/flutter-java.jar.
         > Failed to transform '/Users/Iftach/Desktop/github_sample/flutter-android-jar-sample/android/app/libs/flutter-java.jar' using Jetifier. Reason: IllegalArgumentException, message: Unsupported class file major version 59. (Run with --stacktrace for more details.)
           Suggestions:                                                 
            - Check out existing issues at https://issuetracker.google.com/issues?q=componentid:460323&s=modified_time:desc, it's possible that this issue has already been filed there.
            - If this issue has not been filed, please report it at https://issuetracker.google.com/issues/new?component=460323 (run with --stacktrace and provide a stack trace if possible).
   > Failed to transform flutter-java.jar to match attributes {artifactType=android-classes-jar, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for JetifyTransform: /Users/Iftach/Desktop/github_sample/flutter-android-jar-sample/android/app/libs/flutter-java.jar.
         > Failed to transform '/Users/Iftach/Desktop/github_sample/flutter-android-jar-sample/android/app/libs/flutter-java.jar' using Jetifier. Reason: IllegalArgumentException, message: Unsupported class file major version 59. (Run with --stacktrace for more details.)
           Suggestions:                                                 
            - Check out existing issues at https://issuetracker.google.com/issues?q=componentid:460323&s=modified_time:desc, it's possible that this issue has already been filed there.
            - If this issue has not been filed, please report it at https://issuetracker.google.com/issues/new?component=460323 (run with --stacktrace and provide a stack trace if possible).
```

flutter doctor:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.0.4, on Mac OS X 10.14.6 18G103 darwin-x64, locale en-IL)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] Connected device (2 available)

! Doctor found issues in 1 category.

```

**Comment by TahaTesser** [comment]
@Iftach-Nevo 
It looks like an issue with your own jar file, as mentioned, there is no issue when using Glide jar or Picasso jar. 
Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by Iftach-Nevo** [comment]
@TahaTesser 
Unfortunately this is not an issue with my own jar file. I was able to add my jar file later on (I was doing something wrong before hand).

afterwards I was able to add it to my original project in the android structure window, however when moving back to my flutter project I wasn't able to import the jar file as a library.

<img width="376" alt="Screen Shot 2021-04-27 at 18 16 22" src="https://user-images.githubusercontent.com/58011582/116363258-9dc56c80-a80b-11eb-9f59-d2914d0ffd72.png">

<img width="376" alt="Screen Shot 2021-04-27 at 18 15 05" src="https://user-images.githubusercontent.com/58011582/116363287-a3bb4d80-a80b-11eb-9e65-83dfb780ed6d.png">

<img width="376" alt="Screen Shot 2021-04-27 at 18 14 53" src="https://user-images.githubusercontent.com/58011582/116363306-a8800180-a80b-11eb-8ee9-22cf9185548b.png">

The first image shows that when in Android window the jar is imported fine and I can see my libraries in it, the second image shows that when I go back to my flutter project the jar is closed and the third image shows that it can't be opened as a library and if I do import it my gradle fails building. 
The jar file was built with Android Studio by gradle and is fine.
